File: frontend/src/lib/services/company.ts
import { postJson, putJson } from "@/lib/services/http";
import {Company} from "@/types";

export const companyService = {

  update: (payload: Company) =>
    putJson<{ status: string }>(`/api/v1/companies/${payload.groupId}`, 
      {
        "companyTh": payload.companyTh, 
        "companyEn": payload.companyEn
      }),

  create: (payload: Company) =>
    postJson<{ status: string }>(`/api/v1/companies/${payload.groupId}`, 
      {
        "companyTh": payload.companyTh, 
        "companyEn": payload.companyEn
      }),
};

File: backend/schemas/company.py
from schemas.api import ApiBaseModel


class CompanyInput(ApiBaseModel):
    company_th: str | None = None
    company_en: str | None = None
    
class Company(ApiBaseModel):
    group_id: str
    company_th: str | None = None
    company_en: str | None = None
    aliases: list = []
    
class CompanyList(ApiBaseModel):
    companies: list[Company]